### ISSUE TITLE
Node3D objects will not render when changing scale

### ISSUE THREAD

**Issue by AlanZhouSS** [godotengine/godot#93651]
### Tested versions

v4.3.beta2.official [b75f0485b]

### System information

Godot v4.3.beta2 - macOS 14.5.0 - Vulkan (Mobile) - integrated Apple M1 - Apple M1 (8 Threads)

### Issue description

MeshInstance3D objects will not render when changing SecondNode3D scale.
Code when hidden
```
mesh_instance.scale = Vector3(0.6, 0.6, 0)
```
But when I use the following code, it will render
```
mesh_instance.scale = Vector3(0.6, 0.6, 0.6)
```

I'm not sure if this is correct

### Steps to reproduce

```
extends Node3D

@onready var mesh_instance = $Camera3D/Node3D

func _ready() -> void:
	change_size()

func change_size():
	mesh_instance.scale = Vector3(0.6, 0.6, 0)
```

### Minimal reproduction project (MRP)

Scene tree structure:

![20240627-155554](https://github.com/godotengine/godot/assets/25049229/c965ac97-3e62-4d1d-9032-ff8e7a9a1889)

**Comment by huwpascoe** [comment]
It's child of the camera node so it's probably out of view somehow.

**Comment by Zireael07** [comment]
You set one of the components of scale to 0, of course it won't render

**Comment by huwpascoe** [comment]
> You set one of the components of scale to 0, of course it won't render

It should still render though. I present to you a torus of scale.y = 0
![image](https://github.com/godotengine/godot/assets/1812087/420c613b-801c-4c46-bb35-5a2e262dd566)

**Comment by AlanZhouSS** [comment]
> > You set one of the components of scale to 0, of course it won't render
> 
> It should still render though. I present to you a torus of scale.y = 0 ![image](https://private-user-images.githubusercontent.com/1812087/343723839-420c613b-801c-4c46-bb35-5a2e262dd566.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1NDYxNDksIm5iZiI6MTcxOTU0NTg0OSwicGF0aCI6Ii8xODEyMDg3LzM0MzcyMzgzOS00MjBjNjEzYi04MDFjLTRjNDYtYmIzNS01YTJlMjYyZGQ1NjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjhUMDMzNzI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2VhYWM4Mjc3MmU2NGY5ZGU1MDFkYjcwNjBkNmY0ZTE1MTg1NDExOTIxMmNmZDYxMzc4ZTRlYWI1ZGRiMDY4NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.mIZeOcihGfJpCMaF-ZsyNmfGKobXt9L32Md9slqyQ7g)

Yes, when I create a mesh object in the scene and modify the scale value in the editor interface instead of using a script, the object will be rendered.

**Comment by Mickeon** [comment]
I think there's always been a misunderstanding in how things work here under-the-hood.

> It should still render though. I present to you a torus of scale.y = 0

The reason this renders in the editor is because it knows what the scale is supposed to be _(Node3D caches it)_. When it's time to store on disk, position, rotation, and scale are all stored within `transform` and that's it. Save and reload this exact example, and the scale will be set to `Vector3(0, 0, 0)`. Why? It's literally impossible to get the scale sometimes. This awful note right here tries to explain it:

> "**Note:** Mixed negative scales in 3D are not decomposable from the transformation matrix. Due to the way scale is represented with transformation matrices in Godot, the scale values will either be all positive or all negative."

This note is horrid. The users that can interpret this note don't need to be told. Common users have no idea what this means or how to interpret this information. And obviously it doesn't mention `0.0` at all.


I would frankly consider this a bug, but for different reasons highlighted here. Anytime one of `scale`'s components is `0.0`, the editor _will_ absolutely overwhelm you with the error message `Condition "det == 0" is true` (pretty much _"I cannot know what the scale is"_ ), which I have to assume is because of this line:

https://github.com/godotengine/godot/blob/0c45ace151f25de2ca54fe7a46b6f077be32ba6f/core/math/basis.cpp#L338

So, in a way, the editor already warns you this is a bad idea, but it does not stop you and it's not explicit enough.
